server/controllers/product.js

```diff
@@ -0,0 +1,123 @@
+const express = require('express');
+const bodyParser = require('body-parser');
+const productModel = require('../model/product');
+const Logger = require('../services/logger_services');
+
+const app = express();
+
+const logger = new Logger('app');
+
+app.use(bodyParser.urlencoded({ extended: true }));
+app.use(bodyParser.json());
+
+const newProductValidation = (product) => {
+  if (!product.name) {
+    throw new Error('name field is empty');
+  }
+  if (!(typeof (product.price) === 'number') || !(product.price >= 0)) {
```

Comment:
you can make the if more simple to read:
if (typeof (product.price) !== 'number' || product.price < 0)